From 73ee2b471e3a8043be03e1a9a4e69e4870a32541 Mon Sep 17 00:00:00 2001
From: Grace Cham <hscham@chromium.org>
Date: Wed, 26 Oct 2022 13:45:16 +0900
Subject: [PATCH] Revert "[base/allocator] Support max alloc size with
 use_allocator = "none"."

Revert of r921558
When malloc of more than (or close to) max size is called, it should
return nullptr (and handled by caller) instead of crashing as the
original patch will do.

This reverts commit 324e03c9fba0fd78a721524032eeac43c80dd1db.
---
 ...llocator_shim_default_dispatch_to_glibc.cc | 38 +------------------
 1 file changed, 1 insertion(+), 37 deletions(-)

diff --git a/base/allocator/partition_allocator/shim/allocator_shim_default_dispatch_to_glibc.cc b/base/allocator/partition_allocator/shim/allocator_shim_default_dispatch_to_glibc.cc
index 4f95a3c70c..75fbf696a0 100644
--- a/base/allocator/partition_allocator/shim/allocator_shim_default_dispatch_to_glibc.cc
+++ b/base/allocator/partition_allocator/shim/allocator_shim_default_dispatch_to_glibc.cc
@@ -2,8 +2,6 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include <limits>
-
 #include "base/allocator/partition_allocator/oom.h"
 #include "base/allocator/partition_allocator/partition_alloc_base/compiler_specific.h"
 #include "base/allocator/partition_allocator/partition_alloc_base/numerics/checked_math.h"
@@ -28,28 +26,7 @@ namespace {
 
 using allocator_shim::AllocatorDispatch;
 
-// Strictly speaking, it would make more sense to not subtract amything, but
-// other shims limit to something lower than INT_MAX (which is 0x7FFFFFFF on
-// most platforms), and tests expect that.
-constexpr size_t kMaxAllowedSize = std::numeric_limits<int>::max() - (1 << 12);
-
 void* GlibcMalloc(const AllocatorDispatch*, size_t size, void* context) {
-  // Cannot force glibc's malloc() to crash when a large size is requested, do
-  // it in the shim instead.
-  if (PA_UNLIKELY(size >= kMaxAllowedSize)) {
-    partition_alloc::TerminateBecauseOutOfMemory(size);
-  }
-
-  return __libc_malloc(size);
-}
-
-void* GlibcUncheckedMalloc(const AllocatorDispatch*,
-                           size_t size,
-                           void* context) {
-  if (PA_UNLIKELY(size >= kMaxAllowedSize)) {
-    return nullptr;
-  }
-
   return __libc_malloc(size);
 }
 
@@ -57,11 +34,6 @@ void* GlibcCalloc(const AllocatorDispatch*,
                   size_t n,
                   size_t size,
                   void* context) {
-  const auto total = partition_alloc::internal::base::CheckMul(n, size);
-  if (PA_UNLIKELY(!total.IsValid() || total.ValueOrDie() >= kMaxAllowedSize)) {
-    partition_alloc::TerminateBecauseOutOfMemory(size * n);
-  }
-
   return __libc_calloc(n, size);
 }
 
@@ -69,10 +41,6 @@ void* GlibcRealloc(const AllocatorDispatch*,
                    void* address,
                    size_t size,
                    void* context) {
-  if (PA_UNLIKELY(size >= kMaxAllowedSize)) {
-    partition_alloc::TerminateBecauseOutOfMemory(size);
-  }
-
   return __libc_realloc(address, size);
 }
 
@@ -80,10 +48,6 @@ void* GlibcMemalign(const AllocatorDispatch*,
                     size_t alignment,
                     size_t size,
                     void* context) {
-  if (PA_UNLIKELY(size >= kMaxAllowedSize)) {
-    partition_alloc::TerminateBecauseOutOfMemory(size);
-  }
-
   return __libc_memalign(alignment, size);
 }
 
@@ -111,7 +75,7 @@ size_t GlibcGetSizeEstimate(const AllocatorDispatch*,
 
 const AllocatorDispatch AllocatorDispatch::default_dispatch = {
     &GlibcMalloc,          /* alloc_function */
-    &GlibcUncheckedMalloc, /* alloc_unchecked_function */
+    &GlibcMalloc,          /* alloc_unchecked_function */
     &GlibcCalloc,          /* alloc_zero_initialized_function */
     &GlibcMemalign,        /* alloc_aligned_function */
     &GlibcRealloc,         /* realloc_function */
-- 
2.41.0.rc0.172.g3f132b7071-goog

